src/Microsoft.AspNetCore.Sockets.Common/MessageBatchFormatter.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    public static class MessageBatchFormatter
+    {
+        private const byte BatchTerminator = (byte)';';
+        private const byte TextFormatIndicator = (byte)'T';
+        private const byte BinaryFormatIndicator = (byte)'B';
+
+        public static bool TryFormatMessages(IEnumerable<Message> messages, Span<byte> buffer, MessageFormat format, out int bytesWritten)
```

Comment:
The routine either completely succeeds or fails. Would it be better if it formatted as many messages as it fit in the buffer and then returned the index (or enumerator) of the last one that fit?